atomic_reactor/plugins/post_tag_and_push.py

```diff
@@ -134,18 +137,38 @@ def run(self):
 
                 registry_image = image.copy()
                 registry_image.registry = registry
-                if self.need_skopeo_push():
-                    self.push_with_skopeo(registry_image, insecure, docker_push_secret)
-                else:
-                    self.tasker.tag_and_push_image(self.workflow.builder.image_id,
-                                                   registry_image, insecure=insecure,
-                                                   force=True, dockercfg=docker_push_secret)
-                    defer_removal(self.workflow, registry_image)
+                max_retries = DOCKER_MAX_RETRIES
+
+                expect_v2s2 = False
+                for registry in self.registries:
+                    media_types = self.registries[registry].get('expected_media_types', [])
+                    if MEDIA_TYPE_DOCKER_V2_SCHEMA2 in media_types:
+                        expect_v2s2 = True
+
+                if not (self.group or expect_v2s2):
+                    max_retries = 1
+
+                for retry in range(max_retries):
+                    if self.need_skopeo_push():
+                        self.push_with_skopeo(registry_image, insecure, docker_push_secret)
+                    else:
+                        self.tasker.tag_and_push_image(self.workflow.builder.image_id,
+                                                       registry_image, insecure=insecure,
+                                                       force=True, dockercfg=docker_push_secret)
+
+                    digests = get_manifest_digests(registry_image, registry,
+                                                   insecure, docker_push_secret)
+
+                    if (not (digests.v2 or digests.oci) and (retry + 1 != DOCKER_MAX_RETRIES)):
```

Comment:
Do we need this last part after the `and`? Could it lead to logging a retry that would never happen if max_retries is set to 1?

Reply:
that last part is to prevent logging retry message for last cycle

Reply:
ahh right will change it to != max_retries